src/Umbraco.Core/Services/MemberService.cs

```diff
@@ -894,17 +894,17 @@ public IEnumerable<IMemberGroup> GetAllRoles()
         /// </summary>
         /// <param name="memberId"></param>
         /// <returns>A list of member roles</returns>
-        public IEnumerable<string?>? GetAllRoles(int memberId)
+        public IEnumerable<string> GetAllRoles(int memberId)
         {
             using (ICoreScope scope = ScopeProvider.CreateCoreScope(autoComplete: true))
             {
                 scope.ReadLock(Constants.Locks.MemberTree);
                 var result = _memberGroupRepository.GetMemberGroupsForMember(memberId);
-                return result.Select(x => x.Name).Distinct();
+                return result.Where(x => x.Name is not null).Select(x => x.Name!).Distinct();
```

Comment:
```suggestion
                return result..Select(x => x.Name).WhereNotNull().Distinct();
```
We have an extension method for this 👍

Reply:
Useful, thanks.